src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/magicnumber/InputMagicNumberIgnoreNegativeOctalHex2.java

```diff
@@ -19,15 +19,15 @@ public class InputMagicNumberIgnoreNegativeOctalHex2 {
     /**
      * test long hex
      */
-    long l = 0xffffffffL; // violation
+    long l = 0xffffffffL; // violation ''0xffffffffL' is a magic number'
 
 
     /**
      * test signed values
      */
     public static final int CONST_PLUS_THREE = +3;
     public static final int CONST_MINUS_TWO = -2;
-    private int mPlusThree = +3; // violation
+    private int mPlusThree = +3; // violation ''+3' is a magic number'
```

Comment:
if extra regexp is for this line ..... 

please make it `private int mPlusThree = +3; // violation 'is a magic number'`

message is regexp, and so exact value of number can be skipped, as we do this in all other cases.
We do design test inputs with care and single item per line avoid mentioning full message.

Something like this can escape +, so not ideal but ok apprach if we badly need "+3"
`private int mPlusThree = +3; // violation ''\+3'is a magic number'`